components/provider_cldr/src/transform/timezones/cldr_json/convert.rs

```diff
@@ -0,0 +1,112 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/master/LICENSE ).
+use super::{TimeZoneNames, ZoneFormat};
+use icu_datetime::provider::timezones::{TimeZoneFormatsV1, TimeZoneNameVariantsLongV1, TimeZoneNameVariantsShortV1, TimeZoneNameVariantsV1, TimeZoneNamesLongV1, TimeZoneNamesShortV1};
+use std::collections::BTreeMap;
+
+impl<'d> From<TimeZoneNames> for TimeZoneFormatsV1 {
+    fn from(other: TimeZoneNames) -> Self {
+        Self {
+            hour_format: other.hour_format.into(),
+            gmt_format: other.gmt_format.into(),
+            gmt_zero_format: other.gmt_zero_format.into(),
+            region_format: other.region_format.into(),
+            region_format_variants: other
+                .region_format_variants
+                .into_iter()
+                .map(|(key, value)| (key.into(), value.into()))
+                .collect(),
+            fallback_format: other.fallback_format.into(),
+        }
+    }
+}
+
+impl<'d> From<TimeZoneNames> for TimeZoneNamesLongV1 {
+    fn from(other: TimeZoneNames) -> Self {
+        match other.metazone {
+            None => Self(BTreeMap::new()),
+            Some(metazones) => Self(
+                metazones
+                    .0
+                    .iter()
+                    .filter_map(|(key, metazone)| {
+                        metazone
```

Comment:
Here's another comment that I would have left the first time through, but I didn't want to distract from my other comments.

The metazone identifier, like "Mexico_Pacific", is internal to CLDR. Using that string as the map key in the data provider for symbol lookup is not ideal, in large part because there's no reason to require a heap allocation for what should be an internal identifier string.  I would rather you use the 5-character BCP47 time zone name.  To map from metazone to BCP47 time zone:

1. Map from Metazone to [IANA Golden Zone](https://unicode.org/reports/tr35/tr35-dates.html#Metazones) using metaZones.json.  [Mexico_Pacific's Golden Zone](https://github.com/unicode-org/cldr-json/blob/f321d7704452b076a0f75a988889cfdb87802f35/cldr-json/cldr-core/supplemental/metaZones.json#L6626) is America/Mazatlan.
2. Map from IANA zone to BCP47 zone using [bcp47/timezone.xml](https://github.com/unicode-org/cldr/blob/90aaf561d1827ce06b1e9c4173a70c018665c7d2/common/bcp47/timezone.xml), and you get `"mxmzt"`.

In the end, you get a TinyStr8, which is an ideal type for a map key!  Less space, more efficient, no heap allocations.

Unfortunately, bcp47/timezone.xml is not yet in JSON, but I asked Steven if he can get it in there.

I'd like to see this change before 0.2, but you don't need to prioritize it in this PR if you'd rather make it in a follow-up PR.

Reply:
For the sake of clarity/history:

We decided on 2020-03-04 during our weekly meeting to tentatively keep strings for now until we can have further discussions on what should properly be done here.

The tracking issue is #528